src/main/java/com/google/api/gax/grpc/PageStreamingCallable.java

```diff
@@ -118,5 +132,40 @@ public void remove() {
         throw new UnsupportedOperationException();
       }
     }
+
+    @Override
+    public Iterable<ResourceT> getPageValues() {
+      return pageDescriptor.extractResources(getPage());
+    }
+
+    @Override
+    public PageAccessor<ResourceT> getNextPage() {
+      Object nextToken = getNextPageToken();
+      if (nextToken == null) {
+        return null;
+      } else {
+        RequestT nextRequest =
+            pageDescriptor.injectToken(context.getRequest(), getNextPageToken());
+        return new PageAccessorImpl<>(callable, pageDescriptor,
+                                      context.withRequest(nextRequest));
+      }
+    }
+
+    @Override
+    public String getNextPageToken() {
+      Object nextToken = pageDescriptor.extractNextToken(getPage());
+      if (nextToken.equals(pageDescriptor.emptyToken())) {
+        return null;
+      } else {
+        return nextToken.toString();
+      }
+    }
+
+    private ResponseT getPage() {
+      if (currentPage == null) {
+        currentPage = Futures.getUnchecked(callable.futureCall(context));
```

Comment:
How do you want to throw ApiException instead of UncheckedExecutionException? I tried to use Future.get(Future<V> future, Class<X> exceptionClass), but X could not be a RuntimeException.

The other way is to catch the UncheckedExecutionException and re-throw it as ApiException.